llvm/lib/Target/AArch64/AArch64ISelLowering.h

```diff
@@ -690,6 +690,10 @@ class AArch64TargetLowering : public TargetLowering {
   bool isDesirableToCommuteWithShift(const SDNode *N,
                                      CombineLevel Level) const override;
 
+  bool isDesirableToPullExtFromShl(const MachineInstr &MI) const override {
+    return false;
```

Comment:
A quick comment on the "why" would be good. Could be just the second sentence from the commit message.